src/App.js

```diff
@@ -1,35 +1,73 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm.js";
 
 //custom hook
-const useSemiPersistentState = (initialState, key) => {
-  // Default state to read "savedTodoList" from localStorage
-  const [state, setState] = useState(
-    JSON.parse(localStorage.getItem(key)) || initialState
-  );
-  
-  React.useEffect(() => {
-    // Save todoList to localStorage with the key "savedTodoList"
-    localStorage.setItem(key, JSON.stringify(state));
-    }, [state, key]);
+// const useSemiPersistentState = (initialState, key) => {
+//   // Default state to read "savedTodoList" from localStorage
+//   const [state, setState] = useState(
+//     JSON.parse(localStorage.getItem(key)) || initialState
+//   );
 
-    return [state, setState]
-}
+//   React.useEffect(() => {
+//     // Save todoList to localStorage with the key "savedTodoList"
+//     localStorage.setItem(key, JSON.stringify(state));
+//   }, [state, key]);
+
+//   return [state, setState];
+// };
 
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState([], "savedTodoList");
+  // Default state to read "savedTodoList" from localStorage
+  const initialState = JSON.parse(localStorage.getItem("savedTodoList")) || [];
+  const [todoList, setTodoList] = useState(initialState);
```

Comment:
For this exercise, since you are mocking an API call and having `initialState` set by another function, the initial state should be set to an empty array. So there isn't any confusion about where the state is being set and if it's working correctly.